### DISCUSSION TITLE
Share Topology with V0.6.6

### DISCUSSION THREAD

**Discussion by JonathonMcMullan** [ansys/pyansys-geometry#1468] [Q&A] [answered]
Hi All,

What is the recommended way to use the PrepareTools.share_topology method? I get "self undefined" errors when I attempt to use it based on the documentation

`a = PrepareTools.share_topology(self,bodies=share_bodies,tol=0.001,preserve_instances=False)
print(a)`

Thanks,
Jonathon

**Comment by RobPasMue** [comment]
Hi @JonathonMcMullan!

Here is a script that would show you how to call it. Your main issue is related to the way that the ``PrepareTools`` are called. We (as library developers) need to do a better job at documenting this via examples so I am really sorry about your problem here!

```py
# Launch a Modeler session with your desired product
modeler = launch_modeler()

# Create your design
design = modeler.create_design("MyDesign")

# Perform your modeling operations
...
 
# Perform your request
a = modeler.prepare_tools.share_topology(design.bodies)
print(a)
```

The ``modeler`` object contains an attribute ``prepare_tools`` that gives you access to a properly initialized ``PrepareTools`` object. Hope this helps!

**Reply by JonathonMcMullan** [reply]
Hi @RobPasMue,

Thanks very much for the quick reply. It looks like the method exists which is great but it gives me a version error now. The output claims that it needs 25.1 or newer:

`GeometryRuntimeError: The method 'share_topology' requires a minimum Ansys release version of 25.1.0, but the current version used is 24.2.0.`

Is this correct?

**Reply by RobPasMue** [reply]
Hi @JonathonMcMullan! That is correct, I just checked on my end and this implementation took place on the server sides after the 24R2 release. Meaning that it should be available starting in 25R1. 

However, I did try running it on my end with 24R2 and it seems to work (it might have gotten released in some of the service packs!). Could you give it a try to commenting out in your local installation of PyAnsys Geometry the following?

![image](https://github.com/user-attachments/assets/aa9bbe5e-05a5-40b7-998f-0e022d64e66c)

In file: ``ansys/geometry/core/tools/prepare_tools.py``
Search for "share_topology", and comment out as follows:

```py
    @protect_grpc
    # @check_input_types
    # @min_backend_version(25, 1, 0)
    def share_topology(
        self, bodies: list["Body"], tol: Real = 0.0, preserve_instances: bool = False
    ) -> bool:
```

You just need to comment out these two decorators. Let me know how it goes.

**Reply by RobPasMue** [reply]
Hi @JonathonMcMullan - I just confirmed with our product/service developers that this method is available in 24R2. I also opened #1472 for this purpose. I will do a release today for PyAnsys Geometry that will contain this fix.

**Reply by RobPasMue** [reply, marked as answer]
Hi @JonathonMcMullan - please install ``ansys-geometry-core==0.7.4`` whenever you have the chance. This will allow you to get access for the shared topology on 24R2. Release has just been triggered so it might take some minutes if installation fails on your side.

In order to install it, with your virtual environment active, run the following:

```bash
pip install 'ansys-geometry-core==0.7.4'
```

Let me know if it works!

**Reply by JonathonMcMullan** [reply]
Hi @RobPasMue - thanks for this, updating to 0.7.4 did the trick. I found I needed to explicitly add the "tol" parameter to get it to work but once I did that then it worked without any issues. Thanks for your help!

**Reply by RobPasMue** [reply]
Correct! Tolerance is set to 0 by default. Glad it worked on your side and thanks for marking my response as an answer! This will help other users figure out the solution quicker as well! If you have any questions feel free to open new issues or discussions. =)
